lib/bolt/plugin/aws.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Bolt
+  class Plugin
+    class Aws
+      class EC2
+        attr_writer :client
+
+        def initialize(config)
+          require 'aws-sdk-ec2'
+          @config = config
+          @logger = Logging.logger[self]
+        end
+
+        def name
+          'aws::ec2'
+        end
+
+        def hooks
+          %w[inventory_targets]
+        end
+
+        def config_client(opts)
+          options = {}
+
+          if opts.key?('region')
+            options[:region] = opts['region']
+          end
+          if opts.key?('profile')
+            options[:profile] = opts['profile']
+          end
+          if @config.aws && @config.aws.key?('credentials')
+            creds = File.expand_path(@config.aws['credentials'])
+            if File.exist?(creds)
+              options[:credentials] = ::Aws::SharedCredentials.new(path: creds)
+            end
+          end
+
+          ::Aws::EC2::Client.new(options)
+        end
+
+        def inventory_targets(opts)
+          @client ||= config_client(opts)
```

Comment:
I would move this to the top of `config_client` as `return @client if @client`, and rename that function to just `client`. Then if we need the client elsewhere we won't  accidentally instantiate 2 clients, and the function itself knows if it's already been instantiated.